Blish HUD/GameServices/ArcDps/SocketListener/SocketListener.cs

```diff
@@ -3,141 +3,96 @@
 using System.Net.Sockets;
 using System.Threading;
 
-namespace Blish_HUD.ArcDps
-{
-    public sealed class SocketListener
-    {
-        public delegate void Message(MessageData data);
+namespace Blish_HUD.ArcDps {
+
+    public sealed class SocketListener {
 
         private const int MESSAGE_HEADER_SIZE = 8;
 
         private static readonly Mutex Mutex = new Mutex();
-        private readonly Semaphore _acceptedClientsSemaphore;
+
+        public delegate void Message(MessageData data);
+
         private readonly int _bufferSize;
-        private readonly int _maxConnectionCount;
 
-        private readonly SocketAsyncEventArgsPool _socketAsyncReceiveEventArgsPool;
-        private Socket _listenSocket;
+        private readonly SocketAsyncEventArgs _socketAsyncReceiveEventArgs;
+        private          Socket               _listenSocket;
 
-        public SocketListener(int maxConnectionCount, int bufferSize)
-        {
-            _maxConnectionCount = maxConnectionCount;
+        public SocketListener(int bufferSize) {
             _bufferSize = bufferSize;
-            _socketAsyncReceiveEventArgsPool = new SocketAsyncEventArgsPool();
-            _acceptedClientsSemaphore = new Semaphore(maxConnectionCount, maxConnectionCount);
-
-            for (var i = 0; i < maxConnectionCount; i++)
-            {
-                var socketAsyncEventArgs = new SocketAsyncEventArgs();
-                socketAsyncEventArgs.Completed += OnIoCompleted;
-                socketAsyncEventArgs.SetBuffer(new byte[bufferSize], 0, bufferSize);
-                _socketAsyncReceiveEventArgsPool.Push(socketAsyncEventArgs);
-            }
+
+            var socketAsyncEventArgs = new SocketAsyncEventArgs();
+            socketAsyncEventArgs.Completed += OnIoCompleted;
+            socketAsyncEventArgs.SetBuffer(new byte[bufferSize], 0, bufferSize);
+            _socketAsyncReceiveEventArgs = socketAsyncEventArgs;
         }
 
         public event Message ReceivedMessage;
 
-        public void Start(IPEndPoint localEndPoint)
-        {
-            _listenSocket = new Socket(localEndPoint.AddressFamily, SocketType.Stream, ProtocolType.Tcp)
-            {
+        public void Start(IPEndPoint localEndPoint) {
+            _listenSocket = new Socket(localEndPoint.AddressFamily, SocketType.Stream, ProtocolType.Tcp) {
                 ReceiveBufferSize = _bufferSize
             };
-            _listenSocket.Bind(localEndPoint);
-            _listenSocket.Listen(_maxConnectionCount);
-            StartAccept(null);
+
+            _socketAsyncReceiveEventArgs.AcceptSocket   = _listenSocket;
+            _socketAsyncReceiveEventArgs.RemoteEndPoint = localEndPoint;
+
+            try {
+                if (!_listenSocket.ConnectAsync(_socketAsyncReceiveEventArgs)) ProcessConnect(_socketAsyncReceiveEventArgs);
```

Comment:
Looks like we're just silently failing if the port isn't available for now. Is that correct?

Reply:
Yes, that's right. Should I add a log entry? And which log level?

Reply:
Yeah, a warning level log entry would be appropriate, I think.

Reply:
I added the log, but would like to mention that this will put out a warning every time someone starts gw2 without having the arcdps bridge installed

Reply:
In that case, it may be appropriate to lower the log level to `Info` with something like "Unable to connect to Arcdps-BHUD bridge", but otherwise, that should be fine as it'll help indicate in logs, too, if the bridge was available or not during the session.